### PR TITLE
Feature/getaccounts

### PR DESCRIPTION
**Pull request by RvdK**
creates an acount service

### CODE REVIEW

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTransaction.java [hunk 1]
```diff
@@ -3,52 +3,64 @@
...
+    private Account counterAccount;
+    private Account account;
 
-    @Override
-    // TODO: implement
-    public Direction getDirection() {
-        return null;
+
+    BunqTransaction(PaymentService.ListResponse.Payment payment, Account account) {
+        date = new Date(22111990); //payment.getCreationDate();
```
Comment (Ishadijcks) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTransaction.java]: Why not use payment.getCreationDate()?

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java [hunk 2]
```diff
@@ -60,7 +65,17 @@ public void retrieveRecentTransactions() {
                             "Created session!",
                             Toast.LENGTH_LONG).show();
 
-                    b.listTransactions().thenAccept(ts -> Tools.runOnMainThread(() -> {
+                    Party p = new BunqParty("hello world", 2002);
+                    Account ac = null;
+                    try {
+                        ac = b.listAccount(p).get().stream().findFirst().orElse(new BunqAccount("error", -1, p));
```
Comment (Ishadijcks) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java:71]: Why do you still retrieve the transaction after the account creation has errored?

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java [hunk 3]
```diff
@@ -114,15 +129,17 @@ public void showRecentTransactions(List<? extends Transaction> transactionList)
...
+                details.add("IN");
             }
 //                     TODO: implement once getDate functions correctly
 //                    if (transactionList.get(i).getDate() != null) {
 //                        details.add(transactionList.get(i).getDate().toString());
 //                    }
-            if (transactionList.get(i).getCounterparty() != null) {
-                details.add("" + transactionList.get(i).getCounterparty().toString());
+            if (transactionList.get(i).getCounterAccount() != null) {
+                details.add("" + transactionList.get(i).getCounterAccount().getParty().toString());
```
Comment (Ishadijcks) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java]: Is it possible for `getCounterAccount()` to be not `null`, but `getCounterAccount().getParty()` to be null?

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java [hunk 4]
```diff
@@ -114,15 +129,17 @@ public void showRecentTransactions(List<? extends Transaction> transactionList)
             if (transactionList.get(i).getValue() != null && transactionList.get(i).getCurrency() != null) {
                 details.add(transactionList.get(i).getValue().toString() + " " + transactionList.get(i).getCurrency().toString());
             }
-            if (transactionList.get(i).getDirection() != null) {
-                details.add("" + transactionList.get(i).getDirection().toString());
+            if (transactionList.get(i).getValue() < 0) {
+                details.add("OUT");
+            } else {
+                details.add("IN");
```
Comment (Ishadijcks) [inline BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java]: If a transaction list has a direction, why not `details.add(transactionList.get(i).getDirection())`?

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java (hunk 2 above)
Comment (RvdK) [reply BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java:71]: this is placeholder code and should be refactored into the manager however that is out of scope

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java (hunk 4 above)
Comment (RvdK) [reply BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/activities/RecentTransactionsActivity.java]: i removed the transaction direction as it does not add any usefull information

File: BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTransaction.java (hunk 1 above)
Comment (RvdK) [reply BankChain/app/src/main/java/nl/tudelft/ewi/ds/bankchain/bank/bunq/BunqTransaction.java]: a diffrent bug however it is out of scope of this story

**Review by Ishadijcks — APPROVED** [review]
Approved

**Pull request merged**